### DISCUSSION TITLE
Seeking Assistance with "Run Current File in Interactive Window" Command and Variables window Issues in Theia with Python/Jupyter Extensions

### DISCUSSION THREAD

**Discussion by iokerica** [eclipse-theia/theia#13639] [Q&A]
I have been endeavoring to utilize Theia as a web-based IDE for assisting with Python script development online. To that end, I have installed the following VSCode extensions:

ms-python.python-2024.2.1.vsix
ms-toolsai.jupyter-2024.4.2024041702@linux-x64.vsix
ms-toolsai.jupyter-keymap-1.1.2.vsix
ms-toolsai.jupyter-renderers-1.0.17.vsix
ms-toolsai.vscode-jupyter-cell-tags-0.1.9.vsix
ms-toolsai.vscode-jupyter-slideshow-0.1.6.vsix
vscpde.ipynb-1.83.1.vsix
vscode.builtin-notebook-renderers-1.83.1.vsix
vscode.python-1.83.1.vsix
For the most part, these extensions are functioning as expected within Theia. However, I am encountering two specific issues:

Issue 1: When attempting to right-click on a Python file and select "Run Current File in Interactive Window," an error is generated:

Uncaught (in promise) Error: Command with id 'interactive.open' is not registered.
    at CommandRegistryMainImpl.$executeCommand (command-registry-main.ts:97:19)
    at ProxyHandler.$proxyHandler (proxy-handler.ts:134:71)
    at async RpcProtocol.handleRequest (rpc-protocol.ts:232:28)
Upon delving into the source code of Theia, its plugins, and VSCode itself, it appears that the 'interactive.open' command is indeed exclusive to VSCode, which might explain the error encountered in Theia.

Issue 2: In Jupyter notebooks panel, the Variables window fails to display any variables while executing an ipynb script.

Despite extensive searching within this forum, I have been unable to locate any guidance or solutions addressing these particular problems. Consequently, I am seeking clarification on whether these functionalities are currently supported within Theia, or if there are any known workarounds to overcome these issues. Any assistance or insight would be greatly appreciated.

**Comment by msujew** [comment]
cc @jonah-iden I actually had trouble as well regarding the variable webview. It doesn't show any data after setting a variable in the python kernel. Any recommendations? Aside from that, we should probably implement the `interactive.open` command that is implemented in [VSCode](https://github.com/microsoft/vscode/blob/1867e88862ca26343822231267f20b839daef65d/src/vs/workbench/contrib/interactive/browser/interactive.contribution.ts#L316). It's even documented in [their documentation](https://code.visualstudio.com/api/references/commands#commands), which is pretty unusual.

**Comment by jonah-iden** [comment]
Yeah i havn't figured out the variables view yet. Not quite sure why it doesn't work, but my guess would be that there is still some callback to the jupyter extension somewhere missing.

**Comment by msujew** [comment]
@iokerica FYI with the latest version of the Jupyter VSCode extension (available [here](https://open-vsx.org/extension/ms-toolsai/jupyter)), the variables view is working correctly:

![image](https://github.com/eclipse-theia/theia/assets/4377073/47e459c5-26fd-426a-8f9a-ab4d82fdd051)
